src/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.netcoreapp.cs

```diff
@@ -24,5 +40,162 @@ public static void Create(this DirectoryInfo directoryInfo, DirectorySecurity di
+            {
+                throw new ArgumentOutOfRangeException(nameof(options), SR.ArgumentOutOfRange_Enum);
+            }
+
+            if (bufferSize <= 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(bufferSize), SR.ArgumentOutOfRange_NeedPosNum);
+            }
+
+            SafeFileHandle handle = CreateFileOpenHandle(fileInfo.FullName, mode, rights, share, options, fileSecurity);
+
+            try
+            {
+                return new FileStream(handle, GetFileStreamFileAccess(rights), bufferSize, (options & FileOptions.Asynchronous) != 0);
+            }
+            catch
+            {
+                // If anything goes wrong while setting up the stream, make sure we deterministically dispose of the opened handle.
+                handle.Dispose();
+                throw;
+            }
+        }
+
+        #endregion
+
+        #region Private methods
+
+        // In the context of a FileStream, the only ACCESS_MASK ACE rights we care about are reading/writing data and the generic read/write rights.
+        // See: https://docs.microsoft.com/en-us/windows/win32/secauthz/access-mask
+        private static FileAccess GetFileStreamFileAccess(FileSystemRights rights)
+        {
+            FileAccess access = 0;
+            if ((rights & FileSystemRights.ReadData) != 0 || ((int)rights & GENERIC_READ) != 0)
+            {
+                access = FileAccess.Read;
+            }
+            if ((rights & FileSystemRights.WriteData) != 0 || ((int)rights & GENERIC_WRITE) != 0)
+            {
+                access = access == FileAccess.Read ? FileAccess.ReadWrite : FileAccess.Write;
+            }
+            return access;
+        }
+
+        private static unsafe SafeFileHandle CreateFileOpenHandle(string fullPath, FileMode mode, FileSystemRights rights, FileShare share, FileOptions options, FileSecurity security)
+        {
+            Debug.Assert(fullPath != null);
+
+            // Must use a valid Win32 constant
+            mode = (mode == FileMode.Append) ? FileMode.OpenOrCreate : mode;
+
+            // For mitigating local elevation of privilege attack through named pipes
+            // make sure we always call CreateFile with SECURITY_ANONYMOUS so that the
+            // named pipe server can't impersonate a high privileged client security context
+            // (note that this is the effective default on CreateFile2)
+            int flagsAndAttributes = (int)options | Interop.Kernel32.SecurityOptions.SECURITY_SQOS_PRESENT | Interop.Kernel32.SecurityOptions.SECURITY_ANONYMOUS;
+
+            SafeFileHandle handle;
+
+            fixed (byte* pSecurityDescriptor = security.GetSecurityDescriptorBinaryForm())
+            {
+                Interop.Kernel32.SECURITY_ATTRIBUTES secAttrs = secAttrs = new Interop.Kernel32.SECURITY_ATTRIBUTES
```

Comment:
`secAttrs` = `secAttrs`?
```suggestion
                Interop.Kernel32.SECURITY_ATTRIBUTES secAttrs = new Interop.Kernel32.SECURITY_ATTRIBUTES
```